### PR TITLE
Added Implementation to Cow Navigation - USACO Gold 2017 January

### PR DESCRIPTION
**Pull request by OlexG**

### CODE REVIEW

File: solutions/USACO jan17-cownav.mdx [hunk 1]
```diff
@@ -38,6 +38,157 @@ Don't forget that once Bessie reaches the goal, she will ignore further commands
...
+    int dir1;
+    int x2;
+    int y2;
+    int dir2;
+    int finished1;
+    int finished2;
+    int dist;
+};
+char arr[21][21];
+int visited[21][21][4][21][21][4][2][2];
```
Comment (frodakcin) [inline solutions/USACO jan17-cownav.mdx:83]: You don't actually need this many states; you only care about the two starting states and the "net" direction. When either starting state reaches the end, simply choose to ignore its movement. 

\# of states can be reduced to 20 x 20 x 20 x 20 x 4

Comment (frodakcin) [reply solutions/USACO jan17-cownav.mdx:83]: Whether you want to change this is up to you

File: solutions/USACO jan17-cownav.mdx [hunk 2]
```diff
@@ -38,6 +38,157 @@ Don't forget that once Bessie reaches the goal, she will ignore further commands
...
+                            visited[x][y][d1][z][b][d2][0][0] = -1;
+                            visited[x][y][d1][z][b][d2][1][0] = -1;
+                            visited[x][y][d1][z][b][d2][0][1] = -1;
+                            visited[x][y][d1][z][b][d2][1][1] = -1;
+                        }
+                    }
+                }
+            }
+        }
+    }
```
Comment (frodakcin) [inline solutions/USACO jan17-cownav.mdx:132]: This big chunk can be simply
`memset(visited, -1, sizeof visited);`
or you can use [std::fill](http://www.cplusplus.com/reference/algorithm/fill/)

File: solutions/USACO jan17-cownav.mdx (hunk 1 above)
Comment (frodakcin) [reply solutions/USACO jan17-cownav.mdx:83]: Also you can turn visited into a boolean array

**Review by frodakcin — COMMENTED** [review]
Whether you want to change it is up to you, though

**Comment by frodakcin** [comment]
Requesting michael cuz it's his editorial

File: solutions/USACO jan17-cownav.mdx (hunk 1 above)
Comment (OlexG) [reply solutions/USACO jan17-cownav.mdx:83]: ok sounds good, I will fix it

Comment (OlexG) [reply solutions/USACO jan17-cownav.mdx]: Ok, It has been fixed. I didn't know about the memset function which makes it so much cleaner so thanks for that.

**Review by frodakcin — APPROVED** [review]
looks good!

File: solutions/USACO jan17-cownav.mdx (hunk 1 above)
Comment (frodakcin) [reply solutions/USACO jan17-cownav.mdx]: You're welcome!

**Pull request merged**